Jenkinsfile

```diff
@@ -198,6 +198,30 @@ node('vetsgov-general-purpose') {
     }
   }
 
+  stage('Pre-archive optimizations') {
+    if (shouldBail()) { return }
+
+    try {
+      def builds = [:]
+
+      for (int i=0; i<envNames.size(); i++) {
+        def envName = envNames.get(i)
+
+        builds[envName] = {
+          dockerImage.inside(args) {
+            sh "cd /application && node script/pre-archive/index.js --buildtype=${envName}"
+          }
+        }
+      }
+
+      parallel builds
+    } catch (error) {
+      notify()
+
+      throw error
+    }
+  }
+
```

Comment:
Not sure how I feel about using the word "optimizations", but it seems to fit what's happening here.